### PR TITLE
Doc changes

### PR DESCRIPTION
**Pull request by ltratt**
This PR slightly tweaks the book (https://github.com/softdevteam/grmtools/commit/89592bc21bda66aa55be8f31f65bec340631182f) before starting to update CHANGES.md while I still vaguely remember what we've done (https://github.com/softdevteam/grmtools/commit/1eb16396ad7b8d9f3ceddb5e27bd4a8c81897a72).

@ratmice In theory (if I've done it right!) you should now be able to bors things without me giving you permission each time!

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,33 @@
+# grmtools 0.13.0 (XXXX-XX-XX)
+
+## Breaking changes
+
+* `cfgrammar::yacc::grammar::YaccGrammarSpan` now returns `Option<Span>` rather
+  than `Option<&Span>`.
```
Comment (ratmice) [inline CHANGES.md:6]: I believe this should be:
`cfgrammar::yacc::grammar::YaccGrammar::token_span`,

That function had not made it to a release yet, unfortunately I didn't get the signature right first commit.

So it should perhaps be:
`cfgrammar::yacc::grammar::YaccGrammar::token_span`, was added.

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,33 @@
...
+
+* `cfgrammar::yacc::ast::{Production, Symbol}` no longer derive `Eq` and
+  `PartialEq`. Since both now carry a `Span`, it's easy to confuse "two
```
Comment (ratmice) [inline CHANGES.md:9]: Looks like I missed it in the commit message :frown:, `Symbol` also no longer derives `Hash` for the same reason.

File: CHANGES.md [hunk 3]
```diff
@@ -1,3 +1,33 @@
...
+
+* `Span` has moved from `lrpar` to `cfgrammar`. The import is still available
+  via `lrpar` but is deprecated (though due to
+  https://github.com/rust-lang/rust/issues/30827 this unfortunately does not
+  show as a formal deprecation warning).
+
+## New features
+
+* Improved error messages, with various parts of grammar and lexer files now
+  carrying around `Span` information to help pinpoint errors.
```
Comment (ratmice) [inline CHANGES.md:24]: I believe that is it if we are referring to field additions here, but i'd like to do a more thorough look through the history tomorrow.

File: CHANGES.md (hunk 1 above)
Comment (ltratt) [reply CHANGES.md]: Fixed in 193e052.

File: CHANGES.md (hunk 2 above)
Comment (ltratt) [reply CHANGES.md]: Fixed in 1547ee1.

**Comment by ratmice** [comment]
The signature of `cfgrammar::yacc::GrammarAst::add_rule` changed in a728ac2d55e33bdfcce9d25e341ef402c60e9f5b
to accept a `(name, span): (String, Span)` instead of a `name: String`

`cfgrammar::yacc::GrammarAst::rule_name` was removed, and replaced with `cfgrammar::yacc::GrammarAst::rule_name_str` and `rule_name_span`.  `rule_name_str` has the same signature as the previous `rule_name`, so we may want to consider re-adding it as a deprecated alias?

I think that is everything.

**Comment by ltratt** [comment]
`add_rule` now documented in 4f98c90.

Readded `rule_name` in 2654e95 (which I'll keep as a separate commit if/when we come to squashing).

**Comment by ratmice** [comment]
This looks good to me, Please squash.

**Comment by ltratt** [comment]
Squashed.

**Comment by ratmice** [comment]
I am super confused by the `compare` button, which shows c61a4a970da105e19b645be1d870ab91675f8087,
instead of the diff between 265e95 and c8bd556, but I think this is just the github UI having gone awry.

Manual inspection of the commits all looks ok, so I believe that bors should pick up the right changes.
But didn't want to pull the trigger on this in case you want to double check.

**Comment by ltratt** [comment]
Try doing a force reload of the page? That sometimes fixes these transient errors.

**Comment by ratmice** [comment]
Nope, still showing the wrong diff, not even on this branch.

**Comment by ratmice** [comment]
bors try

[Just want to double check that bors picks up the right commit, probably being overly cautious though]

**Comment by ltratt** [comment]
OK, looks like I didn't get the bors setup right. Bear with.

**Comment by ltratt** [comment]
Try the "try" again?

**Comment by ratmice** [comment]
bors try

**Pull request merged**